models/object_detection_nanodet/NanodetPlus.py

```diff
@@ -0,0 +1,139 @@
+            if hw_scale > 1:
+                newh, neww = self.image_shape[0], int(self.image_shape[1] / hw_scale)
+                img = cv2.resize(srcimg, (neww, newh), interpolation=cv2.INTER_AREA)
+                left = int((self.image_shape[1] - neww) * 0.5)
+                img = cv2.copyMakeBorder(img, 0, 0, left, self.image_shape[1] - neww - left, cv2.BORDER_CONSTANT,
+                                         value=0)  # add border
+            else:
+                newh, neww = int(self.image_shape[0] * hw_scale), self.image_shape[1]
+                img = cv2.resize(srcimg, (neww, newh), interpolation=cv2.INTER_AREA)
+                top = int((self.image_shape[0] - newh) * 0.5)
+                img = cv2.copyMakeBorder(img, top, self.image_shape[0] - newh - top, 0, 0, cv2.BORDER_CONSTANT, value=0)
+        else:
+            img = cv2.resize(srcimg, self.image_shape, interpolation=cv2.INTER_AREA)
+
+        blob = self.pre_process(img)
+        self.net.setInput(blob)
+        outs = self.net.forward(self.net.getUnconnectedOutLayersNames())
+        det_bboxes, det_conf, det_classid = self.post_process(outs)
+        ratioh,ratiow = srcimg.shape[0]/newh,srcimg.shape[1]/neww
+
+        return det_bboxes, left, top, ratioh, ratiow, det_bboxes, det_conf, det_classid
+
+
+    def post_process(self, preds):
+        cls_scores, bbox_preds = preds[::2], preds[1::2]
+        rescale = False
+        scale_factor = 1
+        bboxes_mlvl = []
+        scores_mlvl = []
+        for stride, cls_score, bbox_pred, anchors in zip(self.strides, cls_scores, bbox_preds, self.anchors_mlvl):
+            if cls_score.ndim==3:
+                cls_score = cls_score.squeeze(axis=0)
+            if bbox_pred.ndim==3:
+                bbox_pred = bbox_pred.squeeze(axis=0)
+            bbox_pred = self.softmax_func(bbox_pred.reshape(-1, self.reg_max + 1), axis=1)
+            bbox_pred = np.dot(bbox_pred, self.project).reshape(-1,4)
+            bbox_pred *= stride
+
+            nms_pre = 1000
+            if nms_pre > 0 and cls_score.shape[0] > nms_pre:
+                max_scores = cls_score.max(axis=1)
+                topk_inds = max_scores.argsort()[::-1][0:nms_pre]
+                anchors = anchors[topk_inds, :]
+                bbox_pred = bbox_pred[topk_inds, :]
+                cls_score = cls_score[topk_inds, :]
+
+            points = anchors
+            distance = bbox_pred
+            max_shape=self.image_shape
+            x1 = points[:, 0] - distance[:, 0]
+            y1 = points[:, 1] - distance[:, 1]
+            x2 = points[:, 0] + distance[:, 2]
+            y2 = points[:, 1] + distance[:, 3]
+
+            if max_shape is not None:
+                x1 = np.clip(x1, 0, max_shape[1])
+                y1 = np.clip(y1, 0, max_shape[0])
+                x2 = np.clip(x2, 0, max_shape[1])
+                y2 = np.clip(y2, 0, max_shape[0])
+
+            bboxes = np.stack([x1, y1, x2, y2], axis=-1)
```

Comment:
Try not to call `np.stack` in a loop. It can be too slow.

Reply:
reviewed changes, 
- replaced np.stack